### PR TITLE
Add keepequal and dropequal relabeling config support

### PR DESCRIPTION
**Pull request by cyriltovena**
#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ Main (unreleased)
 
 ### Enhancements
 
+- Add keepequal and dropequal relabeling config support (@ctovena)
```
Comment (rfratto) [inline CHANGELOG.md:42]:
```suggestion
- Flow: Add keepequal and dropequal relabeling config support (@ctovena)
```

**Review by rfratto — COMMENTED** [review]
Can you update the docs for prometheus.relabel, loki.relabel, and discovery.relabel to mention these stages?

File: docs/sources/shared/flow/reference/components/rule-block.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ Here's a list of the available actions, along with a brief description of their
 * `labelmap`  - Matches `regex` against all label names. Any labels that match are renamed according to the contents of the `replacement` field.
 * `labeldrop` - Matches `regex` against all label names. Any labels that match are removed from the metric's label set.
 * `labelkeep` - Matches `regex` against all label names. Any labels that don't match are removed from the metric's label set.
+* `keepequal`: Drop targets for which the concatenated `source_labels` do not match `target_label`.
+* `dropequal`: Drop targets for which the concatenated `source_labels` do match `target_label`.
```
Comment (rfratto) [inline docs/sources/shared/flow/reference/components/rule-block.md:34]: Reformatting to keep with the pattern established by the other bullet points:

```suggestion
* `keepequal` - Drop targets for which the concatenated `source_labels` do not match `target_label`.
* `dropequal` - Drop targets for which the concatenated `source_labels` do match `target_label`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -39,6 +39,8 @@ Main (unreleased)
 
 ### Enhancements
 
+- Flow: Add keepequal and dropequal relabeling config support (@ctovena)
```
Comment (rfratto) [inline CHANGELOG.md:42]: sorry, one more suggestion 😛 

```suggestion
- Flow: Support `keepequal` and `dropequal` actions for relabeling. (@ctovena)
```

**Review by tpaschalis — APPROVED** [review]
Thank you!

**Pull request merged**